[ macOS ] TestWebKitAPI.ResourceLoadStatistics.GrandfatherCallback is a flaky timeout. (256920)

btashawn opened the pull request:
#### f4b4f2f68f7509c97d631ba45ee913616c5f8ee4
<pre>
[ macOS ] TestWebKitAPI.ResourceLoadStatistics.GrandfatherCallback is a flaky timeout. (256920)
<a href="https://bugs.webkit.org/show_bug.cgi?id=256920">https://bugs.webkit.org/show_bug.cgi?id=256920</a>
rdar://109481486

Unreviewed test gardening.

Disabling test while investigated.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/ResourceLoadStatistics.mm:

Canonical link: <a href="https://commits.webkit.org/264185@main">https://commits.webkit.org/264185@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/14deaca849e5bc074d28cdb90d74b9f8269d6dcc

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/6948 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/7181 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/7364 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/8555 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/7186 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/6954 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/8445 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/7119 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/10095 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/7070 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/7743 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/6401 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/8656 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/39/builds/5126 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/6313 "Passed tests") | [⏳ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/GTK-WK2-Tests-EWS "Waiting to run tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/6802 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/6403 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/9230 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/6879 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/5640 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/6252 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/10440 "Built successfully") | | | 
| [❌ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/806 "Failed to push commit to Webkit repository") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/6635 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

ryanhaddad commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/ResourceLoadStatistics.mm:
```diff
@@ -140,7 +144,7 @@ static void ensureITPFileIsCreated()
     TestWebKitAPI::Util::run(&doneFlag);
 }
-
+#endif
```
This needs to move below `TEST(ResourceLoadStatistics, GrandfatherCallback)`

ryanhaddad commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/ResourceLoadStatistics.mm:
```diff
@@ -140,7 +145,6 @@ static void ensureITPFileIsCreated()
     TestWebKitAPI::Util::run(&grandfatheredFlag);
     TestWebKitAPI::Util::run(&doneFlag);
 }
-
```
Please add this newline back before merging.

btashawn committed:
[ macOS ] TestWebKitAPI.ResourceLoadStatistics.GrandfatherCallback is a flaky timeout. (256920)
https://bugs.webkit.org/show_bug.cgi?id=256920
rdar://109481486

Unreviewed test gardening.

Disabling test while investigated.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/ResourceLoadStatistics.mm:

Canonical link: https://commits.webkit.org/264185@main
